snuba/query/data_source/simple.py

```diff
@@ -13,6 +14,7 @@ class Entity(DataSource):
 
     key: EntityKey
     schema: ColumnSet
+    sample_rate: Optional[float] = None
```

Comment:
I think you would have to add this to RelationalSource as well (that's the equivalent of the entity on the physical side of the query).
Also the translator will have to be adapted to propagate this parameter.

Do you plan to remove the sample rate field from the query object as well ?

Reply:
I think sampling is something we should discuss to lock down how it works. I tossed this in here because I couldn't think of another way of saving the sampling rate that is passed in with each entity.